Source/Manager.swift

```diff
@@ -70,16 +73,25 @@ public class Manager: NSURLProtocol {
         }
     }
 
+    /// Remove a resource from the manager
+    static public func removeResource(resource: ResourceProtocol) {
+        resources.remove(resource.resourceIdentifier)
+    }
+
+    /// Remove all resources from the manager
     static public func removeResources() {
         resources.removeAll()
     }
     
     // MARK: - Helpers
     
-    /// Normalize a path which can be used as a Resource Identifier and requested resource ID, if available.
-    /// - Returns: "/path/to/resource/123" -> ("/path/to/resource/", 123)
-    static func resourceNameAndIDFromURL(url: NSURL) -> (name: String?, ids: [Int]?) {
-        var pathComponents = url.pathComponents!
+    /// Normalize a path which can be used as a Resource Identifier and requested resource IDs, if available.
+    /// - Returns: "/path/to/resource/123?foo=bar&baz=quux" -> ("/path/to/resource/", 123, ["foo": ["bar"], "baz": ["quux"])
+    static func resourceArtifactsFromURL(url: NSURL) -> (name: NameArtifact?, ids: IDArtifacts?, query: QueryArtifacts?) {
```

Comment:
How about changing the return type to `(name: NameArtifact, ids: IDArtifacts?, query: QueryArtifacts?)?` (optional tuple with a non-optional name)
If you have a URL, you'll always have a name, if you don't have a URL, you can return nil.